packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+        escrowClient.storeResults(escrowAddress, url, hash)
+      ).rejects.toThrow(ErrorInvalidUrl);
+    });
+
+    test('should throw an error if hash is empty string', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const url = VALID_URL;
+      const hash = '';
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await expect(
+        escrowClient.storeResults(escrowAddress, url, hash)
+      ).rejects.toThrow(ErrorHashIsEmptyString);
+    });
+
+    test('should successfully store results escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const url = VALID_URL;
+      const hash = FAKE_HASH;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await escrowClient.storeResults(escrowAddress, url, hash);
+
+      expect(escrowClient.escrowContract.storeResults).toHaveBeenCalledWith(
+        url,
+        hash
+      );
+    });
+
+    test('should throw an error if the store results fails', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const url = VALID_URL;
+      const hash = FAKE_HASH;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.storeResults.mockRejectedValueOnce(
+        new Error()
+      );
+
+      await expect(
+        escrowClient.storeResults(escrowAddress, url, hash)
+      ).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.storeResults).toHaveBeenCalledWith(
+        url,
+        hash
+      );
+    });
+  });
+
+  describe('complete', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.complete(invalidAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'